tools/walletextension/wallet_extension.go

```diff
@@ -403,15 +409,19 @@ func (we *WalletExtension) handleSubmitViewingKey(userConn userconn.UserConn) {
 
 	vk.SignedKey = signature
 	// create an encrypted RPC client with the signed VK and register it with the enclave
-	// todo (@ziga) - create the clients lazily, to reduce connections to the host.
+	// TODO - Create the clients lazily, to reduce connections to the host.
 	client, err := rpc.NewEncNetworkClient(we.hostAddr, vk, we.logger)
 	if err != nil {
 		userConn.HandleError(fmt.Sprintf("failed to create encrypted RPC client for account %s. Cause: %s", accAddress, err))
 		return
 	}
 	we.accountManager.AddClient(accAddress, client)
 
-	we.persistence.PersistViewingKey(vk)
+	err = we.storage.SaveUserVK(defaultUser, vk)
+	if err != nil {
+		return
```

Comment:
you need to return something here to the user.
using userConn.HandleError

Reply:
added